### PR TITLE
Added OpenGL Core Profile renderer, and some issues fixes.

### LINKED ISSUE #76: Can't build OpenGL sample with MinGW
Issue:

- Operating System (name & version): Windows 10 Professional
- Compiler (name & version): mingw-w64 7.3.0
- Target renderer: OpenGL (Crossplatform)
- cmake command-line used: cmake -DRENDER_OPENGL=On -DGLFW_LIBRARIES=path_to_libglfw3.a -DGLFW_INCLUDE_DIR=path_to_include .

First of all, `CrossPlatform.cpp` does not includes `Windows.h`, because mingw does not set `_MSC_VER` define. 
Whole block from 14 to 22 line can be changed on the following, I guess:
```cpp
#ifdef WIN32
#    include <Windows.h>
#    include <WinBase.h>
#else
#    include <unistd.h>
#endif
#ifdef _MSC_VER
#   define WIN32_LEAN_AND_MEAN
#   undef min
#   undef max
#endif
```

In the second place: When linking OpenGL sample there is an error: 
`localcharset.c:(.text+0xa): undefined reference to `__imp_GetACP'`
That can be fixed by adding `iconv` to linker (-liconv).

### LINKED ISSUE #77: Can't use GWork as cmake `subdirectory`.
Issue:

- Operating System (name & version): Windows 10 Professional
- Compiler (name & version): mingw32-w64 7.3.0
- Target renderer: OpenGL
- cmake command-line used: cmake -DRENDER_OPENGL=On -DGLFW_LIBRARIES=path_to_libglfw3.a -DGLFW_INCLUDE_DIR=path_to_include .

When adding GWork as subdirectory (cmake's `add_subdirectory`), include directories are not appended to target's. To fix this, instead of using `include_directories` it will be more useful to use `target_include_directories(Gwork PUBLIC place_include_directories_here)`.

### PR DESCRIPTION
**Pull request by Megaxela**
- Fixed linkage on Windows mingw. (Resolves #76)
- Replaced `include_directories` to `target_include_directories`. (Resolves #77)
- Added new OpenGL Core Profile renderer (`RENDER_OPENGL_CORE`) with sample. Requires GLEW and GLM. (GLEW - OpenGL extension wrapper) 
- Added ability to remove GLFW as dependency, but OpenGL samples will not build without it. (May be needed if GWork shall be used with another system-specific functional provider, like Qt)
- Modified `README.md` a little.

### CODE REVIEW

File: source/platform/platforms/CrossPlatform.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 #include <Gwork/Platform.h>
 
-#ifdef _MSC_VER
+#ifdef _WIN32
```
Comment (billyquith) [inline source/platform/platforms/CrossPlatform.cpp:14]: Thanks. I think this fixes Mingw?

File: source/platform/renderers/OpenGLCore/OpenGLCore.cpp [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+
+            // Clearing
+            m_activeProgram = 0;
+            m_vertices.clear();
+            glFlush();
+        }
+
+        void OpenGLCore::AddVert(int x, int y, float u, float v)
+        {
+            // OpenGL origin is bottom-left. Gwork origin is top-left.
```
Comment (billyquith) [inline source/platform/renderers/OpenGLCore/OpenGLCore.cpp:378]: I should add this to the docs!

File: source/platform/renderers/OpenGLCore/OpenGLCore.cpp [hunk 3]
```diff
@@ -0,0 +1,743 @@
...
+            auto* data = new unsigned char[iPixelSize * texture->width * texture->height];
+            glGetTexImage(GL_TEXTURE_2D, 0, GL_RGBA, GL_UNSIGNED_BYTE, data);
+            unsigned int iOffset = (y*texture->width+x)*4;
+
+            Color c(data[0+iOffset], data[1+iOffset], data[2+iOffset], data[3+iOffset]);
+
+            // Retrieving the entire texture for a single pixel read
+            // is kind of a waste - maybe cache this pointer in the texture
+            // data and then release later on? It's never called during runtime
+            // - only during initialization.
```
Comment (billyquith) [inline source/platform/renderers/OpenGLCore/OpenGLCore.cpp:516]: I plan to remove pixel sampling at some point. I'd like to have an editor, which just saves to file, and editor has colour picker.

**Review by billyquith — COMMENTED** [review]
Few typo, but tests pass.

**Review by billyquith — APPROVED** [review]

File: source/platform/platforms/CrossPlatform.cpp (hunk 1 above)
Comment (Megaxela) [reply source/platform/platforms/CrossPlatform.cpp:14]: Yep

File: source/platform/renderers/OpenGLCore/OpenGLCore.cpp (hunk 3 above)
Comment (Megaxela) [reply source/platform/renderers/OpenGLCore/OpenGLCore.cpp:516]: You meant theme editor, that will save theme with additional data?

**Pull request merged**